vllm/v1/attention/backends/mla/common.py

```diff
@@ -1570,6 +1570,14 @@ def _run_prefill_new_tokens_trtllm_ragged(
 
         assert prefill.query_seq_lens is not None
         assert prefill.workspace_buffer is not None
+        # allocate BF16 / FP16 output tensor for TRT-LLM ragged attention
+        out = torch.zeros(
```

Comment:
allocating zeros on the hotpath might not be a good idea. Can this be empty? We need to be careful about empty since it can lead to IMAs later

Reply:
got it

Reply:
Acutally it is not necessary for the causal path. Will follow your suggestion- thanks!

Reply:
do you think we should pass in the buffer from the function caller? it looks like there can be buffers provided sometimes

Reply:
size of q is determined during the `generate` call based on the length of prefill, so preallocating and preintializing a buffer of zeros that are later passed into this kernel might need some more work. I'll take that feedback into consideration and revert #30746 instead for the CI fix